src/syscalls/syscall_response.rs

```diff
@@ -64,32 +67,46 @@ impl SyscallResponse {
 //   Response objects
 // ----------------------
 
+/// Gets the timestamp of the block in which the transaction is executed.
 #[derive(Clone, Debug, PartialEq)]
 pub struct GetBlockTimestampResponse {
+    /// The returned timestamp.
     pub timestamp: Felt252,
 }
 
+/// Deploys a new instance of a previously declared class.
 pub struct DeployResponse {
+    /// Address of the deployed contract.
     pub contract_address: Felt252,
+    /// The retdata segment start.
     pub retdata_start: Relocatable,
+    /// The retdata segment end.
     pub retdata_end: Relocatable,
 }
 
+/// Represents error data of any syscall response.
 pub struct FailureReason {
+    /// The retdata segment start.
     pub retdata_start: Relocatable,
+    /// The retdata segment end.
     pub retdata_end: Relocatable,
     // Syscall specific response fields.
     // TODO: this cause circular dependency
     //pub(crate) body: Option<ResponseBody>,
 }
 
+/// Calls a given contract.
 #[derive(Clone, Debug, PartialEq)]
 pub struct CallContractResponse {
```

Comment:
I think here we should write something like "Represents the response from a call_contract syscall" because this struct is that.